modules/ROOT/pages/6.4-release-notes.adoc

```diff
@@ -0,0 +1,1233 @@
+
+As a result, the quickbar no longer opens in {productname} v6.4, when the user is dragging a contenteditable="false" elements.
+
+=== Selection of images, hr, tables or noneditable elements was possible if they were within a noneditable element
+//#TINY-9473
+
+In previous versions of {productname}, `contenteditable="false"` elements that were nested within another such as a element set with the `contenteditable="false"` attribute, would avoid the `contenteditable="false"` guidelines set. As a consequence, selection of images, hrs, tables or `contenteditable="false"` elements was possible if they were nested within a `contenteditable="false"` element.
+
+Improvements were made to check if the DOM parent node is a `contenteditable="false"` element.
+
+In {productname} 6.4, elements with `contenteditable="false"` attributes can no longer be selected when they are children of another `contenteditable="false"` attribute, such as:
+
+.Example of `contenteditable="false"` elements inside `contenteditable="false"` root.
+[source,HTML]
+----
+<div> Parent div element
+  <div contenteditable="false">First child noneditable div element</div>
+  <div contenteditable="true">Second child editable div element</div>
+</div>
+----
+
+=== Ranged deletion of formatted text using selection or keyboard shortcut would sometimes cause Blink and Webkit browsers to insert interpreted tags upon typing, which may result in inconsistent tags
+//#TINY-9302
+
+Blink & Webkit browsers (Chrome & Safari) were deleting the contents of inline formats which would cause the format to be lost and not retained. As a consequence, this issue would impact the semantics and visuals for the *HTML* document, which is not be visible to the user.
+
+A override fix already exists for _character-by-character_ deletion using backspace/delete keys with a collapsed selection. However, this particular bug persists when the format is deleted when there is a ranged selection that starts at the beginning of the inline format(s) and encompasses at least the entirety of the inline format(s). The bug also occurs when deletion is done using common keyboard shortcuts including `Ctrl/Alt + Backspace`, `Ctrl/Alt + Delete`, and `Cmd + Backspace`.
+
+In {productname} 6.4, two new key overrides have been added to:
+
+. Check that a `non-collapsed` selection satisfies the conditions set (specifically for a ranged selection using `backspace/delete`).
+. Check a range of conditions *before* continuing (for keyboard shortcut `backspace/delete`).
+
+As a result, the ranged selection for backspace/delete now check that all of the original formats at the cursor selection *start*, before deletion, are retained with either the native browser behaviour or a caret. For keyboard shortcuts that are using backspace/delete, will create a new unformatted caret which prevents browsers from inserting interpolated formats.
```

Comment:
```suggestion
The native `contenteditable` behaviour on Blink & Webkit browsers (Chrome & Safari) does not retain the original inline format(s) at the caret after all content within a format tag has been deleted. Upon typing again, the browser will attempt to recreate the formats under certain conditions:

. When the selection is collapsed and the last character or element of the content of a format tag is deleted by pressing Backspace or Delete;
. When a ranged selection has been made from the start of the inner content of a format tag and includes the content's entirety, and the ranged selection is deleted by pressing Backspace or Delete; or
. When a keyboard shortcut, including Ctrl+Backspace (Windows/Linux), Ctrl+Delete (Windows/Linux), Opt+Delete (macOS), Opt+Fn+Delete (macOS), and Cmd+Delete (macOS), is used to delete all content within a format tag and the post-deletion caret is at where the start of the content used to be.

The recreated formats are not consistent with the original formats. For example, a `<span style="text-decoration: underline;">` tag will be recreated as a `<u>` tag. This can consequently impact the HTML document's semantics and visuals, causing a bug.

A key override on the Backspace and Delete keys already exists to address the first condition. However, the bug persists for the second and third conditions.

In {productname} 6.4, two new key overrides have been added to address the second and third conditions:

* For Condition 2: When Backspace or Delete is detected on `keydown` on a ranged selection, the override checks if Condition 2 is satisfied. If so, the override will read the current formats, delete the selection, and then create a caret containing the original formats.
* For Condition 3: When one of the keyboard shortcuts from Condition 3 is detected on `keyup`, the override creates an unformatted caret at the current selection. Performing the override on `keyup` allows the native behaviour of the deletion shortcuts to occur. Then, applying an unformatted caret at the selection after deletion will prevent the browser from recreating any formats.
```

Reply:
@mkzhx  Actioned in suggested new changes review.